pkg/executor/aggregate/agg_util.go

```diff
@@ -122,7 +109,7 @@ func GetGroupKey(ctx sessionctx.Context, input *chunk.Chunk, groupKey [][]byte,
 		}
 		expression.PutColumn(buf)
 	}
-	return groupKey, nil
+	return groupKey[:numRows], nil
```

Comment:
Why change this?

Reply:
> Why change this?

Suppose we get block `1` with 2000 rows and block `2` with 1000 rows. Length of groupKey will be 2000 after handling block `1`, then we receive block `2`, if directly return groupKey, the length of groupKey will be 2000, not 1000.